### ISSUE TITLE
Color Contrast Accessibility of Disabled && Checked Checkbox

### ISSUE THREAD

**Issue by keithwyland** [salesforce-ux/design-system#442]
When disabled and checked, the checkbox's color contrast is too low between the background of the box and the color of the checkmark. I propose the checkbox work similar to a disabled checked radio button, where the indicator is blue, but the background remains grey.

Disabled and checked checkbox would be a useful value display for a view screen that contains a boolean read-only value which can be modified via checkbox on a different edit screen.

![image](https://cloud.githubusercontent.com/assets/1273179/25144255/d05f8d4e-2432-11e7-83e7-866b4b66a73e.png)

![image](https://cloud.githubusercontent.com/assets/1273179/25144438/6cf40054-2433-11e7-8763-34f123ddfd9c.png)

**Comment by ishakasliwal** [comment]
Thanks for this proposal @keithwyland. 

@owenschoppe + @SiTaggart, do you have any thoughts on this?

**Comment by SiTaggart** [comment]
I would love to see our disabled styles to get a bit more contrast, if I'm honest ❤️. I know you don't technically have to, but I think it'd be an improvement for low vision users.

**Comment by owenschoppe** [comment]
I thought we submitted an updated style for these already. I don't support
a blue indicator but I can get behind making the indicator the same color
as the border.

Sent from mobile

On Apr 18, 2017, at 2:30 PM, Simon Taggart <notifications@github.com> wrote:

I would love to see our disabled styles to get a bit more contrast, if I'm
honest ❤️. I know you don't technically have to, but I think it'd be an
improvement for low vision users.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/salesforce-ux/design-system/issues/442#issuecomment-294988482>,
or mute the thread

**Comment by keithwyland** [comment]
Here's a comparison of some options. I can understand not wanting to use the enabled blue color. The input border color doesn't provide much better visibility than the existing white. Would the label text color work?

![image](https://cloud.githubusercontent.com/assets/1273179/25179931/94439428-24d1-11e7-8147-bccca34aad5e.png)

**Comment by owenschoppe** [comment]
Nice explorations. 

If you check the contrast going from white to #a8b7c7 moves the needle from 1.3 -> 1.6, but  point taken that it's still subtle. #9faab5 would take us to 1.9 FYI. 

I'd be ok moving it to the #54698d which pushes it all the way to 4.4. It's more clear than the blue that the box is disabled. It is a big jump though.

You should get the opinion of @tquach12 and @Elizabeth-Clarke though before making any changes to the system.

**Comment by owenschoppe** [comment]
The other thing I wanted to add is that we use these same disabled styles
for icon buttons. In those cases it's useful to lighten the icon too to
help differentiate the states. Hence the a8b7c7 suggestion.

Sent from mobile

On Apr 19, 2017, at 5:27 AM, Keith Wyland <notifications@github.com> wrote:

Here's a comparison of some options. I can understand not wanting to use
the enabled blue color. The input border color doesn't provide much better
visibility than the existing white. Would the label text color work?

[image: image]
<https://cloud.githubusercontent.com/assets/1273179/25179931/94439428-24d1-11e7-8147-bccca34aad5e.png>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/salesforce-ux/design-system/issues/442#issuecomment-295250080>,
or mute the thread

**Comment by engai** [comment]
Hi @keithwyland, @owenschoppe has worked on these and a fix should be released with 2.7.0 (Winter '19)

**Comment by engai** [comment]
Hey folks, this has been resolved with https://github.com/salesforce-ux/design-system/commit/8614d1c0c21430bf6be7675978b2d61689b5c040
